### PR TITLE
Feat/fix lint tsc ledger

### PR DESCRIPTION
**Pull request by dawnseeker8**
**Description**
This PR will provide the code fix for all CI lint relative pipeline. Following changes have been made in this PR
1. change to use `KeyringControllerState` to replace old `KeyringState` from '@metamask/keyring-controller'
2. Big code changed  in `engine.ts` to fix KeyringController relative compile issue due to type mismatch and use deprecative type.
3. Fix a reducer `root.ts`  in `test` package due to missing declaration in test.
4. Removed unused lib in package.json which was detected by `yarn depcheck`.
5. add `browserify-zlib` into .`depcheckrc.yml` file to ignore dep check for that library.

### CODE REVIEW

File: app/component-library/components/Pickers/PickerAccount/PickerAccount.types.ts [hunk 1]
```diff
@@ -27,6 +27,11 @@ export interface PickerAccountProps extends Omit<PickerBaseProps, 'children'> {
    * cell account contianer style.
    */
   cellAccountContainerStyle?: ViewStyle;
+
+  /**
+   * Account type label.
+   */
+  accountTypeLabel?: string;
```
Comment (dawnseeker8) [inline app/component-library/components/Pickers/PickerAccount/PickerAccount.types.ts:34]: This type has been used in PickerAccount component too prevent VS code comlain.

File: app/core/Engine.ts [hunk 2]
```diff
@@ -391,7 +394,7 @@ class Engine {
     const phishingController = new PhishingController();
     phishingController.maybeUpdateState();
 
-    const additionalKeyrings = [QRHardwareKeyring, LedgerKeyring];
+    const additionalKeyrings = [QRHardwareKeyring.type, LedgerKeyring.type];
```
Comment (dawnseeker8) [inline app/core/Engine.ts:397]: the KeyringTypes is declared as string[] , therefore, we can't defined directly use object `QRHardwareKeyring`, `LedgerKeyring`

File: package.json [hunk 3]
```diff
@@ -342,7 +342,6 @@
     "rxjs": "^7.5.5",
     "socket.io-client": "^4.5.3",
     "stream-browserify": "1.0.0",
-    "swappable-obj-proxy": "^1.1.0",
```
Comment (dawnseeker8) [inline package.json:345]: redundant libraray was detected by `yarn depcheck`, after remove it and rebuild the app, and do skin tests, and didnt find out any impact.

File: .depcheckrc.yml [hunk 4]
```diff
@@ -40,6 +40,7 @@ ignores:
   - 'react-native-svg-asset-plugin'
   - 'regenerator-runtime'
   - 'rn-nodeify'
+  - 'browserify-zlib'
```
Comment (dawnseeker8) [inline .depcheckrc.yml:43]: this is used to skip the `depcheck` for `browserify-zlib` library which is important for `ethereumjs-util` to work in react native.

File: app/core/Engine.ts [hunk 5]
```diff
@@ -625,11 +628,10 @@ class Engine {
           toHexadecimal(networkController.state.providerConfig.chainId),
         keyringController: {
           signMessage: keyringController.signMessage.bind(keyringController),
-          signPersonalMessage:
-            keyringController.signPersonalMessage.bind(keyringController),
+          signPersonalMessage: (messageParams: PersonalMessageParams) =>
```
Comment (dawnseeker8) [inline app/core/Engine.ts:631]: This is for fixing the type mismatch complain in `yarn lint:tsc`.

**Review by igorms-cons — APPROVED** [review]

**Review by Akaryatrh — APPROVED** [review]

**Pull request merged**